Kernel/Devices/KeyboardDevice.cpp

```diff
@@ -260,6 +282,8 @@ void KeyboardDevice::key_state_changed(u8 raw, bool pressed)
     if (m_client)
         m_client->on_key_pressed(event);
     m_queue.enqueue(event);
+
+    m_is_numpad_key = true;
```

Comment:
What does this flag do? Why is it needed? The name is clearly not accurate, since not every key press is a numpad key :)

Reply:
Well I'm just resetting the flag here but I understand how it could be confusing. I guess I could just do this inside `if (m_num_lock_on && m_is_numpad_key)` to make it a bit less confusing but I'm not sure if there's anything else I could do about it.
I'm not sure of the exact reasoning behind this but I noticed we receive 0x60 to distinguish between keys that are both present on the numpad and only "enabled" when Num Lock is off (for instance delete, home, page up/down etc.) and the same ones present somewhere else on the keyboard that do not depend on anything. It is received before these "non-numpad" keys from some reason so we need to set this flag to know if, for instance numpad delete is pressed and num lock is on so we know to use numpad_map instead of whatever we tried to use originally. 
I hope this makes sense :)

Reply:
Ah, I see. So some scancodes can have a prefix. :)

From reading this page https://www.win.tue.nl/~aeb/linux/kbd/scancodes-1.html
I think the prefix is actually `0xe0` and not `0x60`. We're masking off the top bit (`u8 ch = raw & 0x7f;`)
We should be looking at if `raw == 0xe0` instead.

Also, let's name the member variable something more descriptive, like `m_has_e0_prefix`.
That way, we can remember whether the previously read byte was an `0xe0` prefix, and act differently if so.

Reply:
I see, thank you for looking into that for me. I will make these changes later today.